platforms/software/dependency-management/src/main/java/org/gradle/internal/component/local/model/LocalVariantMetadata.java

```diff
@@ -24,14 +24,16 @@
 import org.gradle.internal.component.model.VariantResolveMetadata;
 import org.gradle.internal.model.CalculatedValue;
 
+import javax.annotation.Nullable;
+
 /**
  * Implementation of {@link VariantResolveMetadata} which allows variant artifacts to be calculated lazily
  * while holding a project lock.
  */
 public final class LocalVariantMetadata extends DefaultVariantMetadata {
     private final CalculatedValue<ImmutableList<LocalComponentArtifactMetadata>> artifacts;
 
-    public LocalVariantMetadata(String name, Identifier identifier, DisplayName displayName, ImmutableAttributes attributes, ImmutableCapabilities capabilities, CalculatedValue<ImmutableList<LocalComponentArtifactMetadata>> artifacts) {
+    public LocalVariantMetadata(String name, @Nullable Identifier identifier, DisplayName displayName, ImmutableAttributes attributes, ImmutableCapabilities capabilities, CalculatedValue<ImmutableList<LocalComponentArtifactMetadata>> artifacts) {
```

Comment:
❗ Since the nullable ID is passed to the superclass, that should be updated for nullability, too.  That field is exposed on a getter, so there might be further consequences of this change.

Reply:
`DefaultVariantMetadata`, the superclass, already has a `@Nullable` annotation on this constructor parameter and the interface already has a `@Nullable` annotation on the getter.

Reply:
My mistake here, I must have opened the wrong class when I looked for this.